Publisher/Sources/DefaultPublisher.swift

```diff
@@ -366,16 +383,18 @@ extension DefaultPublisher {
     
     // MARK: Stop publisher
     private func performStopPublisherEvent(_ event: StopEvent) {
-        if isStopped {
+        if publisherState.isStoppingOrStopped {
             callback(value: Void(), handler: event.resultHandler)
             return
         }
         
+        publisherState = .stopping
+        
         ablyService.close { [weak self] result in
             switch result {
             case .success:
                 self?.locationService.stopUpdatingLocation()
-                self?.isStopped = true
+                self?.publisherState = .stopped
```

Comment:
You're modifying the publisher state outside of the "safe queue thread" in this callback. In this particular situation, that shouldn't cause much trouble because `stopping` and `stopped` state are handled in the same way in the current implementation. However, this can change in the future so I'd say to add another event like "AblyClosedEvent" that will be responsible of changing that state :wink: